modules/openapi-generator/src/main/resources/typescript-fetch/runtime.mustache

```diff
@@ -183,15 +183,19 @@ export function exists(json: any, key: string) {
     return value !== null && value !== undefined;
 }
 
-export function querystring(params: HTTPQuery) {
+export function querystring(params: HTTPQuery, prefix: string = '') {
     return Object.keys(params)
         .map((key) => {
+            const fullKey = prefix + (prefix.length ? `[${key}]` : key);
```

Comment:
maybe it should be configurable whether to use square brackets `[` `]` or e.g. dots `.`?

Reply:
The format I chose matches the `deepObject` style as it is [specified](https://swagger.io/docs/specification/serialization/#query) in OpenAPI 3.0.  
While I have nothing against other formats, that would IMO be a topic to discuss with the OAI.

AFAIK, this generator currently only supports the `form` style with `explode` enabled and no other combination, so there is a general lack of configurability for query parameters.